### PR TITLE
[DOCS] Add configurable password hashing docs

### PR DESCRIPTION
**Pull request by jkakavas**
Adds documentation about the newly introduced configuration option
for setting the password hashing algorithm to be used for the users
cache and for storing credentials for the native and file realm.

### CODE REVIEW

**Comment by lcawl** [comment]
I think it makes sense for the hash algorithm lists appear in the security settings page instead of the Stack Overview.  I've made that change in https://github.com/elastic/elasticsearch/pull/32849/commits/74048ffba7837dcaa15dd49490c759978a7ef975

**Comment by jkakavas** [comment]
>  I think the password-hashing.asciidoc file should just be deleted.

Agreed. I think everything is covered in security-hash-settings.asciidoc and your additions to the configuring-X-realm.asciidoc.

Thank you @lcawl

**Review by lcawl — APPROVED** [review]
LGTM

File: docs/reference/settings/security-hash-settings.asciidoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
+[float]
+[[hashing-settings]]
+==== User cache and password hash algorithms
+
+Certain realms store user credentials in memory. To limit exposure 
+to credential theft and mitigate credential compromise, cached user credentials
```
Comment (jaymode) [inline docs/reference/settings/security-hash-settings.asciidoc]: maybe `To limit exposure to credential theft and mitigate credential compromise, the cache only stored a hashed version of the user credentials`

**Review by jaymode — APPROVED** [review]
LGTM

File: docs/reference/settings/security-hash-settings.asciidoc (hunk 1 above)
Comment (jkakavas) [reply docs/reference/settings/security-hash-settings.asciidoc]: :+1: Thanks, I like that better too

**Review by albertzaharovits — APPROVED** [review]
LGTM

**Pull request merged**